app/controllers/handoff/CorporationTaxDetailsController.scala

```diff
@@ -63,7 +63,7 @@ class CorporationTaxDetailsController @Inject()(val authConnector: PlayAuthConne
       ctAuthorisedHandoff("HO2", requestData) {
         registeredHandOff("HO2", requestData) { _ =>
           handBackService.processCompanyDetailsHandBack(requestData).map {
-            case Success(_) => Redirect(controllers.reg.routes.PPOBController.show)
+            case payload@Success(_) => Redirect(controllers.reg.routes.PPOBController.show).withLang(Lang(payload.get.language))
```

Comment:
I think using pattern matching to extract the inner here would be better so you can avoid the `.get` later
```suggestion
            case Success(payload) => Redirect(controllers.reg.routes.PPOBController.show).withLang(Lang(payload.language))
```